p2p/pkg/preconfirmation/tracker/tracker.go

```diff
@@ -477,6 +477,12 @@ func (t *Tracker) statusUpdater(
 							"txnHash":          task.commitment.Bid.TxHash,
 							"error":            r.Err.Error(),
 						}
+						if task.commitment.BidderAddress != nil {
```

Comment:
nit: This check can be avoided as you are already ignoring the event in deposit manager if these are not correctly populated. The double check can be avoided.